source/core/TDCodecVia.cpp

```diff
@@ -2272,28 +2363,20 @@ VIREO_FUNCTION_SIGNATUREV(UnflattenFromJSON, UnflattenFromJSONParamBlock) // Str
             memset(buffer, 0, topSize);
             arg[0]._paramType->InitData(buffer);
             arg[0]._paramType->CopyData(arg[0]._pData, buffer);
-            parser.ParseData(arg[0]._paramType, arg[0]._pData);
-            if (log.TotalErrorCount() > 0) {
+            error = parser.ParseData(arg[0]._paramType, arg[0]._pData);
+            if ((error && error != kJSONLV_StrictFieldNotFound)) {
                 arg[0]._paramType->CopyData(buffer, arg[0]._pData);
-                if (log.HardErrorCount() == 0)
-                    badNulls = true;
             }
             arg[0]._paramType->ClearData(buffer);
             delete[] buffer;
         }
     }
-    if (_ParamVarArgCount() > 7) {
-        Int32 code = badPath ? kJSONLV_InvalidPath : badNulls ? kJSONLV_TypeMismatch :  kJSONLV_InvalidString;
-        if (badPath || badNulls || log.HardErrorCount() > 0) {
+    if (_ParamVarArgCount() > 7) { // error I/O wired
```

Comment:
I don't think we are properly adding call chain information or the append string as in other calls.